dspace-spring-rest/src/test/java/org/dspace/app/rest/builder/CollectionBuilder.java

```diff
@@ -78,6 +79,26 @@ public Collection build() {
         return collection;
     }
 
+    /**
+    * Creates a TemplateItem for a given collection.
+    * 
+    * @param context DSpace Context
+    * @param collection DSpace Collection
+    */
+    public static Item createTemplateItem(final Context context, final Collection collection) {
+        CollectionBuilder builder = new CollectionBuilder(context);
+        return builder.create_template_item(collection);
+    }
+
+    private Item create_template_item(final Collection collection) {
+        try {
+            Item item = itemService.createTemplateItem(context, collection);
```

Comment:
If we decide to keep this method as is (see comment above) the builder should keep a reference to the created templateItem and remove it in the cleanup. It is only needed if this method is invoked on a "test collection" created outside the test method (that is the case of the basic test structure) otherwise the "wide test collection" will remain altered for the subsequent tests in the class.

Reply:
So, if we proceed with the withTemplateItem() the template item should be an attribute of the created collection and cease to exist upon the cleanup of the aforementioned collection. Right?